src/libc/mod.rs

```diff
@@ -0,0 +1,323 @@
+    while *cs != 0 {
+        cs = cs.offset(1);
+        count += 1;
+    }
+    count
+}
+
+#[inline(always)]
+pub unsafe fn fcntl(fd: c_int, cmd: c_uint, arg: c_int) -> c_int {
+    linux::fcntl(fd, cmd, arg as c_ulong) as c_int
+}
+
+#[inline(always)]
+pub unsafe fn open64(pathname: *const c_char, flags: c_int, mode: c_int) -> c_int {
+    linux::open(pathname, flags | O_LARGEFILE, mode as linux::umode_t)
+}
+
+#[inline(always)]
+pub unsafe fn readlink(pathname: *const c_char,
+                       buf: *mut c_char,
+                       bufsiz: size_t)
+    -> ssize_t
+{
+    linux::readlink(pathname,
+                    buf,
+                    cmp::min(bufsiz, <c_int>::max_value() as size_t) as c_int)
+}
+
+#[inline(always)]
+pub unsafe fn fchmod(fd: c_int, mode: mode_t) -> c_int {
+    linux::fchmod(fd, mode as linux::umode_t)
+}
+
+#[inline(always)]
+pub unsafe fn chmod(filename: *const c_char, mode: mode_t) -> c_int {
+    linux::chmod(filename, mode as linux::umode_t)
+}
+
+#[inline(always)]
+pub unsafe fn mkdir(pathname: *const c_char, mode: mode_t) -> c_int {
+    linux::mkdir(pathname, mode as linux::umode_t)
+}
+
+#[inline(always)]
+pub unsafe fn lseek64(fd: c_int, offset: off64_t, whence: c_uint) -> off64_t {
+    let mut result_offset: off64_t = 0;
+    let result = linux::_llseek(fd, offset, &mut result_offset, whence);
+    if result >= 0 {
+        result_offset
+    } else {
+        result as off64_t
+    }
+}
+
+#[derive(Clone, Copy)]
+pub struct pthread_attr_t {
+    stack_size: usize,
+}
+
+struct thread {
+    // Required, because one cannot easily read out register containing the
+    // pointer to this structure on some platforms.
+    this: *mut thread,
+    thread_id: pid_t,
+}
+
+#[derive(Clone, Copy)]
+pub struct pthread_t {
+    thread: *mut thread,
+}
+
+pub unsafe fn pthread_attr_init(attr: *mut pthread_attr_t) -> c_int {
+    *attr = pthread_attr_t {
+        stack_size: 0,
+    };
+    0
+}
+
+pub unsafe fn pthread_attr_destroy(attr: *mut pthread_attr_t) -> c_int {
+    pthread_attr_init(attr)
+}
+
+pub unsafe fn pthread_attr_setstacksize(attr: *mut pthread_attr_t,
+                                        stacksize: size_t)
+    -> c_int
+{
+    (*attr).stack_size = stacksize;
+    0
+}
+
+/*
+pub unsafe fn pthread_attr_getstack(attr: *const pthread_attr_t,
+                                    stackaddr: *mut *mut c_void,
+                                    stacksize: *mut size_t)
+    -> c_int
+{
+    *stackaddr = ptr::null_mut();
+    *stacksize = (*attr).stack_size;
+    0
+}
+
+pub unsafe fn pthread_attr_getguardsize(attr: *const pthread_attr_t,
+                                        guardsize: *mut size_t)
+    -> c_int
+{
+    *guardsize = 0;
+    0
+}
+
+pub unsafe fn pthread_getattr_np(pthread: pthread_t,
+                                 attr: *mut pthread_attr_t)
+    -> c_int
+{
+    pthread_attr_init(attr)
+}
+*/
+
+pub unsafe fn pthread_create(pthread: *mut pthread_t,
```

Comment:
It's a simplified version of the MUSL one, `pthread_create` in `src/thread/pthread_create.c`.